ElliotForWater/elliotforwater.com
# added recaptcha

### Related Issue
Resolves #47 

## Description
Added Google Recaptcha v2 to subscribe forms.


#### Impacted Areas in Application
With a valid Recaptcha site key, verification will show:
![![image](https://user-images.githubusercontent.com/47090434/146627043-60274807-0aa7-464c-a014-b41b6ad57d4b.png)](https://user-images.githubusercontent.com/47090434/146627034-f705b536-f90f-46bd-b8e5-801c18a459e4.png)



Without Recaptcha verification, form will not submit and an error toast will show:
![image](https://user-images.githubusercontent.com/47090434/146627049-b29afed5-0f1a-44b6-b5a3-44a376e9c422.png)



When form is submitted successfully, the recaptcha token is also sent with the form data:
![image](https://user-images.githubusercontent.com/47090434/146627114-9a123899-ebbd-4beb-993d-f5c56661dbf9.png)


Docs for verifying token on the server: https://developers.google.com/recaptcha/docs/verify


#### Steps to Test or Reproduce
Sign into google and get a site key. Make sure to add 'localhost' as a domain: https://www.google.com/recaptcha/admin/create


Replace 'FAKE KEY' with your site key:
![image](https://user-images.githubusercontent.com/47090434/146627491-98af2e5a-2bb8-48bb-9c3d-629df6c5a0b9.png)

Alternatively, we could add the Site key to our .env.development file


#### Related PRs
n/a

------------------------------------------------
## PR Checklist:
